code/modules/mining/machinery/material_extractor.dm

```diff
@@ -0,0 +1,248 @@
+// Input is clockwise east, output clockwise west, gas output clockwise south
+// The general concept is that it heats up things up to 20c and grabs anything gaseous or liquid at that temp range, with a debuff on liquids, so it doesn't replace more specialized machines.
+#define GAS_EXTRACTOR_GAS_TANK 1000
+#define GAS_EXTRACTOR_REAGENTS_TANK 500
+#define GAS_EXTRACTOR_EXTRACTION_RATE 0.5 //units per tick
+#define GAS_EXTRACTOR_OPERATING_TEMP T20C + 1 //Temperature the machine heat stuff to.. Has to be 20c + 1 because someone decided ice melted at 20c
+#define GAS_EXTRACTOR_LIQUID_EFFICIENCY 0.75 //% efficiency for liquids
+#define GAS_EXTRACTOR_WAIT_TIME_DUMP 1 SECONDS //Wait at least that many seconds after inserting something for the machine to dump the result. So reactions happens
+
+//TODO: Sprite work?
+
+/obj/machinery/atmospherics/unary/material/extractor
+	name = "Gas extractor"
+	desc = "A machine for extracting liquids and gases from ices and hydrates."
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "extractor"
+	layer = MOB_LAYER+1 // Overhead
+	density = 1
+
+	obj_flags = OBJ_FLAG_ANCHORABLE | OBJ_FLAG_ROTATABLE
+	use_power = POWER_USE_OFF
+	idle_power_usage = 25
+	power_rating = 1 KILOWATTS
+	power_channel = EQUIP
+	connect_types = CONNECT_TYPE_REGULAR
+
+	uncreated_component_parts = null
+	construct_state = /decl/machine_construction/default/panel_closed
+	required_interaction_dexterity = DEXTERITY_SIMPLE_MACHINES
+	maximum_component_parts = list(/obj/item/stock_parts = 10, /obj/item/chems = 1)
+
+	var/const/max_items = 25 //Max amount of items it can process at the same time
+	var/time_last_input = 0 //Time since an item was processed, used to wait a bit for reactions to happen
+	var/time_last_bark = 0 //time since the machine said anything last
+	var/obj/item/chems/output_container //#TODO: change this when plumbing is a thing
+	var/global/list/items_whitelist = list( //stuff that should be accepted
+		/obj/item/ore,
+	)
+
+/obj/machinery/atmospherics/unary/material/extractor/Initialize(mapload, d = 0, populate_parts = TRUE)
+	. = ..()
+	if(populate_parts)
+		output_container = new/obj/item/chems/glass/bucket(src)
+	air_contents.volume = GAS_EXTRACTOR_GAS_TANK
+	if(!reagents)
+		create_reagents(GAS_EXTRACTOR_REAGENTS_TANK)
+	QUEUE_TEMPERATURE_ATOMS(src)
+
+/obj/machinery/atmospherics/unary/material/extractor/Bumped(var/obj/O)
+	//Ignore if we can't function
+	if(inoperable())
+		return
+	
+	//Sanity filter
+	if(O.pass_flags > PASS_FLAG_TABLE || O.anchored)
+		return
+
+	//Only handles entities touching the machine from the input's direction only
+	if(get_dir(loc, O.loc) == get_input_dir())
+		//try processing
+		if(process_ore(O))
+			time_last_input = world.time
+		else
+			//Reject stuff that we cannot process
+			O.dropInto(get_output_loc())
+			if(world.time > (time_last_bark + 4 SECONDS))
+				time_last_bark = world.time
+				state("\The [O.name] was rejected for processing!")
```

Comment:
This should probably be silent or have a delay, otherwise it'll get kinda spammy if you accidentally put something like ore into it.

Reply:
Oh, nevermind.